src/test/kotlin/com/stringconcat/tdd/MoneyTest.kt

```diff
@@ -3,81 +3,157 @@ package com.stringconcat.tdd
+        Money.dollar(BigDecimal.valueOf(5)) shouldBe Money.dollar(BigDecimal.valueOf(5))
     }
 
     @Test
     fun `5 USD is NOT equal to 7 USD`() {
-        Money.dollar(5) shouldNotBe Money.dollar(7)
+        Money.dollar(BigDecimal.valueOf(5)) shouldNotBe Money.dollar(BigDecimal.valueOf(7))
     }
 
     @Test
     fun `No dollar is equal to null`() {
-        Money.dollar(5).shouldNotBe(null)
+        Money.dollar(BigDecimal.valueOf(5)).shouldNotBe(null)
     }
 
     @Test
     fun `5 USD multiply 2 should be 10 USD`() {
-        Money.dollar(5) * 2  shouldBe Money.dollar(10)
+        Money.dollar(BigDecimal.valueOf(5)) * 2 shouldBe Money.dollar(BigDecimal.valueOf(10))
     }
 
     @Test
     fun `10 USD multiply 3 should be 30 USD`() {
-        Money.dollar(10) * 3  shouldBe Money.dollar(30)
+        Money.dollar(BigDecimal.valueOf(10)) * 3 shouldBe Money.dollar(BigDecimal.valueOf(30))
     }
 
     @Test
     fun `2 USD + 2 USD is 4 USD`() {
-        Money.dollar(2) + Money.dollar(2) shouldBe Money.dollar(4)
+        Money.dollar(BigDecimal.valueOf(2)) + Money.dollar(BigDecimal.valueOf(2)) shouldBe Money.dollar(
+            BigDecimal.valueOf(
+                4
+            )
+        )
     }
 
     @Test
     fun `2 USD is not equal 2 CHF`() {
-        Money.dollar(2) shouldNotBe Money.franc(2)
+        Money.dollar(2.toBigDecimal()) shouldNotBe Money.franc(2.toBigDecimal())
+    }
+
+    @Test
+    fun `5 USD to CHF is 15 CHF if rate 3 to 1`() {
+        Money.dollar(BigDecimal.valueOf(5)).convert(3.toBigDecimal(), Money.Currency.CHF) shouldBe Money.franc(
+            BigDecimal.valueOf(15)
+        )
+    }
+
+    @Test
+    fun `15 CHF to USD is 5 USD if rate 1 to 3`() {
+        Money.franc(BigDecimal.valueOf(15)).convert(
+            1.toBigDecimal().divide(3.toBigDecimal(), 4, RoundingMode.HALF_EVEN),
+            Money.Currency.USD
+
+        ) shouldBe Money.dollar(BigDecimal.valueOf(5))
     }
 
     @Test
     fun `5 USD to CHF is 10 CHF if rate 2 to 1`() {
-        Money.dollar(5).toFranc() shouldBe Money.franc(10)
+        Money.dollar(BigDecimal.valueOf(5)).convert(2.toBigDecimal(), Money.Currency.CHF) shouldBe Money.franc(
+            BigDecimal.valueOf(10)
+        )
     }
 
     @Test
     fun `5 CHF to CHF is 5 CHF`() {
-        Money.franc(5).toFranc() shouldBe Money.franc(5)
+        Money.franc(BigDecimal.valueOf(5))
+            .convert(1.toBigDecimal(), Money.Currency.CHF) shouldBe Money.franc(5.toBigDecimal())
     }
+
+    @Test
+    fun `10 CHF to USD is 5 USD if rate 1 to 2`() {
+        Money.franc(BigDecimal.valueOf(10)).convert(
+
+            1.toBigDecimal().divide(2.toBigDecimal(), 4, RoundingMode.HALF_EVEN),
+            Money.Currency.USD
+
+        ) shouldBe Money.dollar(BigDecimal.valueOf(5))
+    }
+
+    @Test
+    fun `10 CHF + 5 USD to USD is 10 USD if CHFUSD rate 1 to 2`() {
+        Money.franc(BigDecimal.valueOf(10)).convert(0.5.toBigDecimal(), Money.Currency.USD) + Money.dollar(
```

Comment:
1. Мне кажется такая запись шумновата (здесь и в похожих местах). Не удалось сходу понять что выполняется в тесте. Вот как бы я записал:

```suggestion
      val franc = Money.franc(BigDecimal.valueOf(10))
      val dollar = Money.dollar(BigDecimal.valueOf(5))
      val rate = 0.5.toBigDecimal()

      val result = franc.convert(rate, Money.Currency.USD) + dollar

      result shouldBe Money.dollar(10.toBigDecimal())
```

На мой взгляд немного понтнее. 

2. Другой момент - непонятно как родился такой тест и что он проверяет. Мы вроде как отдельно должны проверить как конвертится, отдельно как складывается. Тут я не могу сказать о чем на самом деле тест. Но о том, как и что должно быть в тестах мы ещё поговорим, можно пока забить